19 set up a cicd pipeline for triggering tests

Anjali0407-git opened the pull request:
**Fixes #19**

kungfuchicken reviewed (changes requested):
Having separate workflows for front and back makes sense if you use a filter under `branches:` for component based on the path name. https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#using-filters

Anjali0407-git commented:
That really makes sense. I addressed your feedback by adding `paths` filter with either `server/` or `client/` path along with `github/workflows/` path

Anjali0407-git commented:
I pushed CODEOWNERS update to this PR as it's a very small change.
